src/generator/pollers.ts

```diff
@@ -74,15 +77,34 @@ func (p *${poller.name}) Poll(ctx context.Context) (*${responseType}, error) {
 	return result, nil
 }
 
+// Wait will continue to poll until a terminal state is reached or an error is encountered. Wait will use the 
+// duration specified in the retry-after header, if the header is not specified then the pollingInterval that
+// is specified will be used to wait between polling requests. 
 func (p *${poller.name}) Wait(ctx context.Context, pollingInterval time.Duration) (*${responseType}, error) {
-	return nil, nil
+	for {
+		resp, err := p.Poll(context.Background())
+		if err != nil {
+			return nil, err
+		}
+		if p.Done() {
+			return resp, err
+		}
+		if ra := resp${rawResponse}.Header.Get(azcore.HeaderRetryAfter); len(ra) > 0 {
```

Comment:
You can use `resp.RetryAfter()` which has all this code in it already.

Reply:
SO I had the same thought and tried to do that until I realized that the response that we receive here is the response envelope for the operation.

Reply:
Per our conversation can use `p.response()`.

Reply:
Done